src/WarcraftLegacies.Source/Quests/Frostwolf/QuestThunderBluff.cs

```diff
@@ -1,53 +1,59 @@
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using MacroTools;
 using MacroTools.Extensions;
 using MacroTools.FactionSystem;
 using MacroTools.ObjectiveSystem.Objectives;
 using MacroTools.QuestSystem;
 using WarcraftLegacies.Source.Setup.Legends;
+using WCSharp.Shared.Data;
 using static War3Api.Common;
 
 
 namespace WarcraftLegacies.Source.Quests.Frostwolf
 {
+  /// <summary>
+  /// Kill the centaur leader in Mulgore and bring <see cref="LegendFrostwolf.LegendCairne"/> to <see cref="LegendFrostwolf.LegendThunderbluff"/> to unlock it
+  /// </summary>
   public sealed class QuestThunderBluff : QuestData
   {
     private readonly List<unit> _rescueUnits = new();
 
-    public QuestThunderBluff(PreplacedUnitSystem preplacedUnitSystem, rect rescueRect) : base("The Long March",
+    /// <inheritdoc />
+    public QuestThunderBluff(PreplacedUnitSystem preplacedUnitSystem, Rectangle rescueRect) : base("The Long March",
       "The Tauren have been wandering for too long. The plains of Mulgore would offer respite from this endless journey.",
       "ReplaceableTextures\\CommandButtons\\BTNCentaurKhan.blp")
     {
-      AddObjective(new ObjectiveUnitIsDead(preplacedUnitSystem.GetUnit(FourCC("ncnk"), Regions.ThunderBluff.Center)));
-      AddObjective(new ObjectiveAnyUnitInRect(Regions.ThunderBluff, "Thunder Bluff", true));
+      AddObjective(new ObjectiveUnitIsDead(preplacedUnitSystem.GetUnit(FourCC("ncnk"), rescueRect.Center)));
+      AddObjective(new ObjectiveLegendInRect(LegendFrostwolf.LegendCairne, rescueRect, "Thunder Bluff"));
```

Comment:
This is a positive lore change, but I think it negatively narrows down options. Players should be allowed to refrain from building a hero without losing the opportunity to get one of their primary main bases.

If we want to expand on the flavour of the quest, I would recommend modifying the CompletionPopup property so that it displays different flavour based on whether it's Cairne or some other hero that completes it.